packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,173 @@
+   */
+  onRequestClose: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  title: null,
+  message: null,
+  actions: [],
+  variant: variants.CUSTOM,
+};
+
+const actionSection = (actions) => {
+  let actionButton = null;
+  let dismissButton = null;
+  if (!actions || !actions.length) {
+    return null;
+  }
+
+  for (let i = 0; i < actions.length; i += 1) {
+    const action = actions[i];
+    if (action.key === 'action') {
+      actionButton = <Button text={action.text} variant={Button.Opts.Variants.EMPHASIS} onClick={action.onClick} />;
+    }
+    if (action.key === 'dismiss') {
+      dismissButton = <Button text={action.text} onClick={action.onClick} />;
+    }
+  }
+  return <div className={cx('actions')}>{actionButton}{dismissButton}</div>;
+};
+
+const getIcon = (variant, customIcon = null) => {
+  switch (variant) {
+    case variants.ALERT:
+      return (<IconAlert />);
+    case variants.ERROR:
+      return (<IconError />);
+    case variants.WARNING:
+      return (<IconWarning />);
+    case variants.ADVISORY:
+      return null;
+    case variants.INFO:
+      return (<IconInformation />);
+    case variants.SUCCESS:
+      return (<IconSuccess />);
+    case variants.CUSTOM:
+      return customIcon;
+    default:
+      return null;
+  }
+};
+
+class NotificationDialog extends React.Component {
+  render() {
+    if (!this.props.isOpen) {
+      return null;
+    }
+
+    const {
+      header,
+      title,
+      message,
+      actions,
+      variant,
+      customIcon,
+      isOpen,
+      onRequestClose,
+      ...customProps
+    } = this.props;
+
+    const defaultHeader = variant === variants.CUSTOM ? '' : `${variant}`;
+    const notificationDialogClassNames = cx('notification-dialog', customProps.className);
+
+    return (
+      <AbstractModal
+        ariaLabel="NotificationDialog"
+        classNameModal={notificationDialogClassNames}
+        isOpen={this.props.isOpen}
+        onRequestClose={this.props.onRequestClose}
+        closeOnEsc={false}
+        closeOnOutsideClick={false}
+        zIndex="8000"
+      >
+        <div className={cx('header-body')}>{header || defaultHeader}</div>
+        <div className={cx('notification-dialog-body')}>
+          {variant &&
+            <div className={cx('icon-div')}>{getIcon(variant, customIcon)}</div>
+          }
+          <div className={cx('notification-dialog-body-text')}>
+            <div className={cx('title')}>{title}</div>
+            <div className={cx('message')}>{message}</div>
+          </div>
+        </div>
+        <div className={cx('footer-body')}>{actionSection(actions)}</div>
+      </AbstractModal>
+    );
+  }
+}
+
+NotificationDialog.propTypes = propTypes;
+NotificationDialog.defaultProps = defaultProps;
+NotificationDialog.Opts = {};
+NotificationDialog.Opts.Variants = variants;
+
+export default NotificationDialog;
```

Comment:
This 'controlled' NotificationDialog allows for a lot of flexibility in the component's presentation. In addition to this, I think we should provide an 'uncontrolled' version that can be presented synchronously and managed outside of the calling component. This would be helpful to consumers who want to inject the notification dialog into Promise chains and control logic flow with the action selections.

Something like the confirmAlert function as seen here: https://github.com/GA-MO/react-confirm-alert/blob/master/src/index.js

I'm not sure if this needs to happen with the initial PR, but I do think it would be helpful.

Reply:
https://github.com/cerner/terra-framework/issues/177
Logged an issue for now. I think this would be a good enhancement issue rather than pushing changes in the initial PR.